orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionRunnerSupport.java

```diff
@@ -37,99 +45,154 @@ public ExecutionRunnerSupport(Collection<StageDefinitionBuilder> stageDefinition
   }
 
   /**
-   * The default implementation plans all stages.
-   * Implementations may override this method completely or use it to plan the entire execution before running it.
-   *
-   * @param execution the execution instance with skeleton stages.
-   * @param <T>       the execution type.
-   * @throws Exception
-   */
-  @Override
-  public <T extends Execution<T>> void start(T execution) throws Exception {
-    List<Stage<T>> stages = new ArrayList<>(execution.getStages()); // need to clone because we'll be modifying the list
-    stages.stream().forEach(this::planStage);
-  }
-
-  /**
    * Plans the tasks in a stage including any pre and post stages.
-   * Implementations may call this directly before executing an individual stage or all in advance.
+   * Implementations may call this directly before executing an individual stage
+   * or all in advance.
    *
-   * @param stage the stage with no tasks currently attached.
-   * @param <T>   the execution type.
+   * @param stage    the stage with no tasks currently attached.
+   * @param callback callback invoked with the tasks of the stage.
+   * @param <T>      the execution type.
    */
-  protected <T extends Execution<T>> void planStage(Stage<T> stage) {
+  protected <T extends Execution<T>> void planStage(
+    Stage<T> stage,
+    // TODO: surely we can just use the 2nd consumer form here and handle single-element
```

Comment:
Yeah, this is kind of the one shitty thing I want to clean up before merging this.